app/controllers/users/totp_setup_controller.rb

```diff
@@ -0,0 +1,46 @@
+module Users
+  class TotpSetupController < ApplicationController
+    before_action :confirm_two_factor_authenticated
+
+    def new
+      if session[:otp_secret_key].nil?
+        session[:otp_secret_key] = current_user.generate_totp_secret
+      end
```

Comment:
For now, we need to add `:otp_secret_key` to list of keys that gets deleted when the user signs out here: https://github.com/18F/identity-idp/blob/master/app/controllers/concerns/saml_idp_logout_concern.rb#L137-L141

In addition, that session key should be removed when the user confirms their TOTP. Otherwise, they could disable TOTP, then enable it again, and end up with same `otp_secret_key`. Also, if a user on a shared computer enables TOTP, then closes their browser, I believe the next person could sign in and enable TOTP and will get the same codes as the other user.

I think it's a good idea to delete session keys that are no longer needed right away.